tests/src/test/scala/org/http4s/UriSpec.scala

```diff
@@ -57,16 +58,49 @@ class UriSpec extends Http4sSpec with MustThrownMatchers {
       }
 
       "parse port correctly" >> {
-        "if there is one" in {
-          val uri = getUri("http://localhost:8080/")
-          uri.port must_=== Some(8080)
-        }
+        "if there is a valid (non-negative) one" >> prop { nonNegative: Int =>
+          {
+            val uri = getUri(s"http://localhost:$nonNegative/")
+            uri.port must_=== Some(nonNegative)
+          }
+        }.setGen(Gen.choose[Int](0, Int.MaxValue))
         "if there is none" in {
           val uri = getUri("http://localhost/")
           uri.port must_=== None
         }
       }
 
+      "fail to parse port" >> {
+        "if it's negative" >> prop { negative: Int =>
+          {
+            val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$negative/")
+            uri match {
+              case Left(ParseFailure("Invalid URI", _)) => ok
+              case unexpected => ko(unexpected.toString)
+            }
+          }
+        }.setGen(Gen.choose[Int](Int.MinValue, -1))
+        "if it's larger than Int.MaxValue" >> prop { tooBig: Long =>
+          {
+            val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$tooBig/")
+            uri match {
+              case Left(ParseFailure("Invalid URI", _)) => ok
+              case unexpected => ko(unexpected.toString)
+            }
+          }
+        }.setGen(Gen.choose[Long]((Int.MaxValue: Long) + 1, Long.MaxValue))
+        "if it's not a number, but not an empty String" >> prop { notNumber: String =>
+          {
+            val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$notNumber/")
+            uri match {
+              case Left(ParseFailure("Invalid URI", _)) => ok
+              case unexpected => ko(unexpected.toString)
+            }
+          }
+        }.setGen(Gen.alphaNumStr.suchThat(str => str.nonEmpty && Try(str.toInt).isFailure))
+
+      }
```

Comment:
Are we testing empty strings after a colon anywhere?  Those are irritatingly legal.

Reply:
Good point, Ross. I'll add such a test.